api/py/ai/chronon/pyspark/executables.py

```diff
@@ -0,0 +1,764 @@
+from __future__ import annotations
+
+import copy
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta
+from typing import Any, Generic, TypeVar, cast
+
+from py4j.java_gateway import JVMView, JavaObject
+from pyspark.sql import DataFrame, SparkSession
+
+from ai.chronon.api.ttypes import (
+    GroupBy, Join, JoinPart, JoinSource, Query, Source, StagingQuery
+)
+from ai.chronon.pyspark.constants import PARTITION_COLUMN_FORMAT
+from ai.chronon.repo.serializer import thrift_simple_json
+from ai.chronon.utils import (
+    __set_name as set_name, get_max_window_for_gb_in_days, output_table_name
+)
+
+# Define type variable for our executables
+T = TypeVar('T', GroupBy, Join)
```

Comment:
@camweston-stripe should we also include StagingQuery and add a staging query executable? 
```suggestion
T = TypeVar('T', GroupBy, Join, StagingQuery)
```